app/jobs/process_delius_data_job.rb

```diff
@@ -11,28 +11,40 @@ class ProcessDeliusDataJob < ApplicationJob
 
   include ApplicationHelper
 
-  def perform(nomis_offender_id)
+  # identifier_type can be :nomis_offender_id (default), or :crn
+  def perform(identifier, identifier_type: :nomis_offender_id)
     ApplicationRecord.transaction do
-      logger.info("[DELIUS] Processing delius data for #{nomis_offender_id}")
-
-      DeliusImportError.where(nomis_offender_id: nomis_offender_id).destroy_all
-
-      import_data(nomis_offender_id)
+      logger.info("#{identifier_type}=#{identifier},job=process_delius_data_job,event=started")
+      import_data(identifier, identifier_type: identifier_type)
+      logger.info("#{identifier_type}=#{identifier},job=process_delius_data_job,event=finished")
     end
   end
 
 private
 
-  def import_data(nomis_offender_id)
-    probation_record = OffenderService.get_probation_record(nomis_offender_id)
+  def import_data(identifier, identifier_type:)
+    probation_record = OffenderService.get_probation_record(identifier)
 
     if probation_record.nil?
       return logger.error(
-        "nomis_offender_id=#{nomis_offender_id},job=process_delius_data_job,event=missing_probation_record|" \
+        "#{identifier_type}=#{identifier},job=process_delius_data_job,event=missing_probation_record|" \
         'Failed to retrieve probation record'
       )
     end
 
+    if identifier_type == :crn
+      nomis_offender_id = probation_record.fetch(:noms_id)
+
+      if nomis_offender_id.blank?
+        return logger.error(
+          "crn=#{identifier},job=process_delius_data_job,event=missing_offender_id|" \
+          'Probation record does not have a NOMIS ID'
+        )
```

Comment:
That's quite confusing and not self-documenting - does this mean we're using the value we return?

Let's keep it clearer:

```
        logger.error(
          "crn=#{identifier},job=process_delius_data_job,event=missing_offender_id|" \
          'Probation record does not have a NOMIS ID'
        )
        return
```